Board.java

```diff
@@ -57,4 +59,8 @@ public ArrayList availablePositions() {
         return positions;
     }
 
+    public boolean isGameOver() {
+        return getState().equals("Game Over");
+    }
```

Comment:
There is something about this state checking on a String that I don't like.
For one, what if you were to change the String value?  You'd need to change it in several places.  You could fix this with a constant or an enum, but still something would irk me..
It feels like isGameOver should actually be doing the check for a win rather relying on the state having been set correctly somewhere else.. 
checkForWin(player1) && checkForWin(player2) could be called instead, plus one other check to see if there are no availableSpaces left.